django/template/engine.py

```diff
@@ -90,6 +104,15 @@ def template_context_processors(self):
         context_processors += tuple(self.context_processors)
         return tuple(import_string(path) for path in context_processors)
 
+    def get_template_builtins(self, builtins):
+        return [import_library(x) for x in builtins]
+
+    def get_template_libraries(self, libraries):
+        loaded = {}
+        for name, path in libraries.items():
+            loaded[name] = import_library(path)
+        return loaded
+
     @cached_property
```

Comment:
I don't remember anything in particular, I just refactored incrementally and ended up there.